### PR TITLE
CMake: Add uninstall target for base library and Python bindings.

### PR DESCRIPTION
**Pull request by AlexandraTrifan**
Signed-off-by: Alexandra.Trifan <Alexandra.Trifan@analog.com>

### CODE REVIEW

File: bindings/python/CMakeLists.txt [hunk 1]
```diff
@@ -85,5 +85,5 @@ configure_file(${SETUP_PY_IN} ${SETUP_PY})
 add_custom_target(${PROJECT_NAME}py ALL DEPENDS _${PROJECT_NAME} WORKING_DIRECTORY ${CMAKE_BINARY_DIR} COMMAND ${PYTHON_EXECUTABLE_LIB_VERSION} ${SETUP_PY} build)
 
 if(NOT SKIP_INSTALL_ALL)
-	install(CODE "execute_process(WORKING_DIRECTORY ${CMAKE_BINARY_DIR} COMMAND ${PYTHON_EXECUTABLE_LIB_VERSION} ${SETUP_PY} install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX})")
+	install(CODE "execute_process(WORKING_DIRECTORY ${CMAKE_BINARY_DIR} COMMAND ${PYTHON_EXECUTABLE_LIB_VERSION} ${SETUP_PY} install --record=${CMAKE_BINARY_DIR}/install_manifest_py.txt --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX})")
```
Comment (commodo) [inline bindings/python/CMakeLists.txt]: can you break this on multiple lines?

File: CMakeLists.txt [hunk 2]
```diff
@@ -254,3 +254,8 @@ if (NOT OSX_PACKAGE)
 		endif()
 	endif()
 endif()
+
+
```
Comment (commodo) [inline CMakeLists.txt]: nitpick: extra line

File: cmake/Modules/uninstall.cmake [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        )
+        
+        if(NOT "${result}" STREQUAL 0)
+            message(FATAL_ERROR "Failed to remove file: '${file}'.")
+        endif()
+    else()
+        MESSAGE(STATUS "File '${file}' does not exist.")
+    endif()
+
+
```
Comment (commodo) [inline cmake/Modules/uninstall.cmake]: nitpick: extra line

File: cmake/Modules/uninstall.cmake [hunk 4]
```diff
@@ -0,0 +1,80 @@
+set(MANIFEST "${CMAKE_BINARY_DIR}/install_manifest.txt")
```
Comment (commodo) [inline cmake/Modules/uninstall.cmake]: curios: is this `uninstall.cmake` a custom/internal file, or is it referenced from somewhere else?

File: cmake/Modules/uninstall.cmake [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+if(NOT EXISTS ${MANIFEST})
+    message(FATAL_ERROR "Cannot find install manifest: '${MANIFEST}'")
+endif()
+
+file(STRINGS ${MANIFEST} files)
```
Comment (commodo) [inline cmake/Modules/uninstall.cmake]: would it be easier to add all files into a single `files` variable and iterate over it once?

File: cmake/Modules/uninstall.cmake (hunk 4 above)
Comment (AlexandraTrifan) [reply cmake/Modules/uninstall.cmake]: It's a custom module and it should only be used by the top directory CMakeLists.txt when uninstalling.

File: cmake/Modules/uninstall.cmake (hunk 5 above)
Comment (AlexandraTrifan) [reply cmake/Modules/uninstall.cmake]: done

**Pull request merged**